src/ui/components/Rating/index.js

```diff
@@ -49,8 +49,26 @@ export class RatingBase extends React.Component {
     this.props.onSelectRating(rating);
   }
 
+  // Helper function used to render title attributes
+  // for each individual star, as well as the wrapper
+  // that surrounds the read-only set of stars
+  renderTitle = (i18n, rating, readOnly, starRating) => {
```

Comment:
This could be cleaned up a bit. For one, you could remove all the `else if` statements, because each `if` returns something, so there's no need for any `else`s.

You could also try grouping it a bit, for example:

```js
  renderTitle = (i18n, rating, readOnly, starRating) => {
    if (readOnly) {
      if (rating) {
        return i18n.sprintf(i18n.gettext('Rated %(rating)s out of 5.'),
          { rating: i18n.formatNumber(parseFloat(rating).toFixed(1)) });
      }
      return i18n.gettext('This add-on has not been rated yet.');
    }
    if (rating) {
      return i18n.sprintf(i18n.gettext(
        `Update your rating to %(starRating)s out of 5.`), { starRating });
    }
    return i18n.sprintf(i18n.gettext(
      `Rate this add-on %(starRating)s out of 5.`), { starRating });
  }
```